src/clients/inst_autopost.rb

```diff
@@ -47,21 +50,18 @@ def main
       )
       @progress_stages = [_("Install required packages")]
 
-      @steps = Builtins.size(Builtins.filter(Y2ModuleConfig.ModuleMap) do |p, d|
-        (Ops.get_string(d, "X-SuSE-YaST-AutoInst", "") == "all" ||
-          Ops.get_string(d, "X-SuSE-YaST-AutoInst", "") == "write") &&
-          Builtins.haskey(
-            Profile.current,
-            Ops.get_string(d, "X-SuSE-YaST-AutoInstResource", p)
-          )
-      end)
+      registry = Y2Autoinstallation::Entries::Registry.instance
+      modules_to_write = registry.writable_descriptions.select do |description|
```

Comment:
I am not sure whether `writable_descriptions` is a good name.

Reply:
I am sure more names are not ideal, but I do not find better. I welcome any suggestions